### DISCUSSION TITLE
How can i change token from database in flask

### DISCUSSION THREAD

**Discussion by hiddify-com** [eternnoir/pyTelegramBotAPI#1934] [Q&A] [answered]
I want to create a bot with flask. My bot token is exist in the database and can be changed via an api.
Thereforem how can i change the bot token in runtime?

**Comment by Badiboy** [comment]
No need to change token. Just create new instance of TeleBot.

**Reply by hiddify-com** [reply]
But the @bot.message_handler will be called with the older token 

```
bot = telebot.TeleBot(API_TOKEN)
def change_bot(new_token):
     bot=telebot.TeleBot(new_token)
# Handle '/start' and '/help'
@bot.message_handler(commands=['help', 'start'])
def send_welcome(message):
    bot.reply_to(message,
                 ("Hi there, I am EchoBot.\n"
                  "I am here to echo your kind words back to you."))
```

**Reply by hiddify-com** [reply]
It would be great if we can call `bot = telebot.TeleBot()`
and then we can call `bot.set_token(TOKEN)`

It is useful when we want to lazy loading the bot or change the bot token

For example now we can not initialize bot without token and since we save the token in the database, it is not accessible from the begining ;)

**Comment by hiddify-com** [comment, marked as answer]
Just need to create a new Bot with empty token
``` 
bot = telebot.TeleBot("") 
```
Then change bot token:
```
def register(new_token):
     bot.token=new_token
```
